tangelo/linq/target/target_qulacs.py

```diff
@@ -80,8 +81,92 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
+                    measurements += measure
+                    success_probability *= cprob
+                    if desired_meas_result:
+                        del dmeas[0]
+
+                    # If a CMEASURE has occured
+                    if cmeasure_flags[0] is not None:
+                        if isinstance(cmeasure_flags[0], str):
+                            newcirc = source_circuit.controlled_measurement_op(measure)
+                        elif isinstance(cmeasure_flags[0], dict):
+                            newcirc = Circuit(cmeasure_flags[0][measure], n_qubits=source_circuit.width)
+                        new_unitary_circuits, new_qubits, new_cmeasure_flags = get_unitary_circuit_pieces(newcirc)
+                    # No classical control
+                    else:
+                        new_unitary_circuits = [Circuit(n_qubits=source_circuit.width)]
+                        new_qubits = []
+                        new_cmeasure_flags = []
+
+                    del unitary_circuits[0]
+                    del qubits[0]
+                    del cmeasure_flags[0]
+                    del precirc[0]
+                    precirc[0] = new_unitary_circuits[-1] + precirc[0]
+
+                    if len(new_unitary_circuits) > 1:
+                        unitary_circuits = new_unitary_circuits[:-1] + unitary_circuits
+                        qubits = new_qubits + qubits
+                        cmeasure_flags = new_cmeasure_flags + cmeasure_flags
+                        precirc = [Circuit()]*len(qubits) + precirc
+
+                source_circuit._probabilities[measurements] = success_probability
+                applied_gates += precirc[0]._gates + unitary_circuits[-1]._gates
+                source_circuit._applied_gates = applied_gates
+                translated_circuit = translate_c(precirc[0]+unitary_circuits[-1], "qulacs", output_options={"save_measurements": True})
+                state.load(sv)
+                translated_circuit.update_quantum_state(state)
+                self._current_state = state.get_vector()
+                python_statevector = self._current_state
+                if self.n_shots is None:
+                    frequencies = self._statevector_to_frequencies(self._current_state)
+                    for meas, val in frequencies.items():
+                        self.all_frequencies[measurements + meas] = val
+                else:
+                    bitstr = self._int_to_binstr(state.sampling(1)[0], source_circuit.width)
+                    samples[measurements+bitstr] = samples.get(measurements+bitstr, 0) + 1
+
+                # Call the finalize method of ClassicalControl, used to reset variables, perform computation etc.
+                source_circuit.finalize_cmeasure_control()
+
+            if self.n_shots:
+                self.all_frequencies = {k: v / self.n_shots for k, v in samples.items()}
+                frequencies = {k: v / self.n_shots for k, v in samples.items()}
```

Comment:
I thinl L165 could be deleted. it recomputes the same thing as L164 and you're not using the variable for the return statement right after.

Reply:
I do not understand why this was resolved and why L188 couldn't be deleted. The value of that variable is not used in the return statement hit by the code on the next line, and will be lost.